Watchman.Discord.UnitTests/Users/UsersControllerTests.cs

```diff
@@ -0,0 +1,109 @@
+
+        [Test]
+        public async Task GetAvatar_ShouldDisplayUserAvatar()
+        {
+            //Arrange
+            var command = new AvatarCommand() { User = 1};
+            var userContext = testContextsFactory.CreateUserContext(1, "test");
+            var contexts = testContextsFactory.CreateContexts(1, 1, 1);
+
+            var messagesServiceMock = new Mock<IMessagesService>();
+            var messagesServiceFactoryMock = new Mock<IMessagesServiceFactory>();
+            messagesServiceFactoryMock.Setup(x => x.Create(It.IsAny<Contexts>()))
+                .Returns(messagesServiceMock.Object);
+            var usersServiceMock = new Mock<IUsersService>();
+            usersServiceMock.Setup(x => x.GetUserByIdAsync(It.IsAny<DiscordServerContext>(), It.IsAny<ulong>()))
+                .Returns<DiscordServerContext, ulong>((a, b) => Task.FromResult(userContext));
+
+            var controller = this.testControllersFactory.CreateUsersController(
+                usersServiceMock: usersServiceMock,
+                messagesServiceFactoryMock: messagesServiceFactoryMock);
+               
+            //Act
+            await controller.GetAvatar(command, contexts);
+
+            //Assert
+            usersServiceMock.Verify(x => x.GetUserByIdAsync(contexts.Server, command.User), Times.Once);
+            messagesServiceMock.Verify(x => x.SendMessage(userContext.AvatarUrl, It.IsAny<MessageType>()), Times.Once);
+        }
+
+        [Test]
+        public async Task GetAvatar_ShouldRespondThatUserHasNoAvatar()
+        {
+            //Arrange
+            var command = new AvatarCommand();
+            var userContext = testContextsFactory.CreateUserContext(1);
+            var contexts = testContextsFactory.CreateContexts(1, 1, 1);
+
+            var messagesServiceMock = new Mock<IMessagesService>();
+            messagesServiceMock.Setup(x => x.SendMessage(It.IsAny<string>(), It.IsAny<MessageType>()))
+               .Returns<string, MessageType>((a, b) => Task.FromResult(userContext.AvatarUrl));
+            var messagesServiceFactoryMock = new Mock<IMessagesServiceFactory>();
+            messagesServiceFactoryMock.Setup(x => x.Create(It.IsAny<Contexts>()))
+                .Returns(messagesServiceMock.Object);
+            var usersServiceMock = new Mock<IUsersService>();
+            usersServiceMock.Setup(x => x.GetUserByIdAsync(It.IsAny<DiscordServerContext>(), It.IsAny<ulong>()))
+                .Returns(Task.FromResult(userContext));
+
+            var controller = this.testControllersFactory.CreateUsersController(
+                usersServiceMock: usersServiceMock,
+                messagesServiceFactoryMock: messagesServiceFactoryMock);
+
+            //Act
+            await controller.GetAvatar(command, contexts);
+
+            //Assert
+            usersServiceMock.Verify(x => x.GetUserByIdAsync(contexts.Server, command.User), Times.Never);
```

Comment:
możesz tu wrzucić sprawdzanie It.IsAny
bo chyba nie sprawdzamy w ogóle

Reply:
Tak w zasadzie te Verify z GetUserByIdAsync mógłby wyrzucić całkowicie z testów, bo to nieistotne jakimi drogami uzyskaliśmy dany UserContext, tylko istotne jest to, że uzyskaliśmy jakiś porządany efekt (w tym przypadku interesuje nas tylko fakt, że wyświetlono komunikat o braku avatara)

Reply:
w tym konkretnym teście to może nie ma sensu, bo tak samo to powinno zadziałać niezależnie czy pobieramy swojego awatara czy cudzego

ale w przypadku innych testów, chcemy mieć potwierdzenie czy działamy na swoim czy cudzym UserContext